zio-http/jvm/src/test/scala/zio/http/ServerSentEventSpec.scala

```diff
@@ -39,8 +44,124 @@ object ServerSentEventSpec extends ZIOHttpSpec {
+    suite("::encode")(
+      test("single string data") {
+        val event  = ServerSentEvent(data = "hello world")
+        val result = event.encode
+        assertTrue(
+          result ==
+            """data: hello world
+              |
+              |""".stripMargin,
+        )
+      },
+      test("multiline string data") {
+        val event  = ServerSentEvent(data = "line1\nline2\nline3")
+        val result = event.encode
+        assertTrue(
+          result ==
+            """data: line1
+              |data: line2
+              |data: line3
+              |
+              |""".stripMargin,
+        )
+      },
+      test("data with eventType") {
+        val event  = ServerSentEvent(data = "test data", eventType = Some("message"))
+        val result = event.encode
+        assertTrue(
+          result ==
+            """event: message
+              |data: test data
+              |
+              |""".stripMargin,
+        )
+      },
+      test("data with id") {
+        val event  = ServerSentEvent(data = "test data", id = Some("123"))
+        val result = event.encode
+        assertTrue(
+          result ==
+            """data: test data
+              |id: 123
+              |
+              |""".stripMargin,
+        )
+      },
+      test("data with retry") {
+        val event  = ServerSentEvent(data = "test data", retry = Some(5000.milliseconds))
+        val result = event.encode
+        assertTrue(
+          result ==
+            """data: test data
+              |retry: 5000
+              |
+              |""".stripMargin,
+        )
+      },
+      test("all fields") {
+        val event  = ServerSentEvent(
+          "test data",
+          eventType = Some("update"),
+          id = Some("456"),
+          retry = Some(3000.milliseconds),
+        )
+        val result = event.encode
+        assertTrue(
+          result ==
+            """event: update
+              |data: test data
+              |id: 456
+              |retry: 3000
+              |
+              |""".stripMargin,
+        )
+      },
+      test("eventType with newlines gets flattened") {
+        val event  = ServerSentEvent(data = "data", eventType = Some("message\nwith\nnewlines"))
+        val result = event.encode
+        assertTrue(
+          result ==
+            """event: message with newlines
+              |data: data
+              |
+              |""".stripMargin,
+        )
+      },
+      test("id with newlines gets flattened") {
+        val event  = ServerSentEvent(data = "data", id = Some("id\nwith\nnewlines"))
+        val result = event.encode
+        assertTrue(
+          result ==
+            """data: data
+              |id: id with newlines
+              |
+              |""".stripMargin,
+        )
+      },
+      test("empty string data") {
+        val event  = ServerSentEvent(data = "")
+        val result = event.encode
+        assertTrue(result == "data: \n\n")
```

Comment:
This PR changes how the empty `data` SSE event is sent. 
Before, it was sending an empty string while now, it sends this `data: ` empty event.
AFAIU the spec (https://html.spec.whatwg.org/multipage/server-sent-events.html), that's what is expected (but I can be wrong. The spec is a bit obscure)